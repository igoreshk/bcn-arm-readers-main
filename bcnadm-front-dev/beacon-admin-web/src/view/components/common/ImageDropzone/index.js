import React from 'react';
import PropTypes from 'prop-types';
import useImageDropzone from 'utils/useImageDropzone';

const ImageDropzone = ({ translate, input, imageLink, maxSize }) => {
  const image = input.value;
  const onDropRejectedCallback = () => {
    input.onChange(null);
  };
  const onDropAcceptedCallback = (selectedFile) => {
    input.onChange(selectedFile);
  };

  const { file, isRejected, errorMessage, getRootProps, getInputProps } = useImageDropzone({
    translate,
    image,
    maxSize,
    onDropRejectedCallback,
    onDropAcceptedCallback
  });

  return (
    <div {...getRootProps({ className: 'dropzone' })}>
      <input {...getInputProps()} />
      {!image && !imageLink && <span>{translate('imageDropzone.upload')}</span>}
      {(imageLink || image || file) && <img className="preview" src={file || image || imageLink} />}
      {isRejected && <span className="error">{errorMessage}</span>}
    </div>
  );
};

ImageDropzone.propTypes = {
  input: PropTypes.object,
  imageLink: PropTypes.string,
  translate: PropTypes.func.isRequired,
  maxSize: PropTypes.number.isRequired
};

export default ImageDropzone;
