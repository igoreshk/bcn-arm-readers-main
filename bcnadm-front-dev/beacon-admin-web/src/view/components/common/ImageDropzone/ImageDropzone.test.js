import ImageDropzone from 'src/view/components/common/ImageDropzone/index';
import React from 'react';
import { shallow } from 'enzyme';

const props = {
  imageLink: '/building/1/image',
  input: {
    value: {
      image: ''
    }
  },
  maxSize: 100,
  translate: jest.fn()
};

const wrapper = shallow(<ImageDropzone {...props} />);

describe('ImageDropzone test', () => {
  it('check props matches', () => expect(wrapper.length).toEqual(1));
  it('snapshot for ImageDropzone', () => expect(wrapper).toMatchSnapshot());
});
