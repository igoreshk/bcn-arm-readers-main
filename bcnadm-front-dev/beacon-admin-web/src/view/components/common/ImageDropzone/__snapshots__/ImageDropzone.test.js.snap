// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ImageDropzone test snapshot for ImageDropzone 1`] = `
<div
  className="dropzone"
  onBlur={[Function]}
  onClick={[Function]}
  onDragEnter={[Function]}
  onDragLeave={[Function]}
  onDragOver={[Function]}
  onDrop={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  tabIndex={0}
>
  <input
    accept=".png, .jpeg, .jpg, .svg"
    autoComplete="off"
    multiple={true}
    onChange={[Function]}
    onClick={[Function]}
    style={
      Object {
        "display": "none",
      }
    }
    tabIndex={-1}
    type="file"
  />
  <img
    className="preview"
    src={
      Object {
        "image": "",
      }
    }
  />
</div>
`;
